cli/flox/src/commands/mod.rs

```diff
@@ -524,6 +526,14 @@ pub enum EnvironmentSelect {
     Unspecified,
 }
 
+#[derive(Debug, Error)]
+pub enum EnvironmentSelectError {
+    #[error(transparent)]
+    Environment(#[from] EnvironmentError2),
+    #[error(transparent)]
+    Anyhow(#[from] anyhow::Error),
```

Comment:
Nit: these variants provide very little context.
We use them to capture `DotFloxNotFond` but _only_ iff unable to discover.
`DotFloxNotFound` for a provided dir falls kind of in the wrong bucket here too.

I think the relevant context for this error is:
1) Failed to discover a dot flox
1*) Failed to find a dot flox where specified (this might just as well be one single variant)
2) Found environment but could not open
3) Failed to find upstream remote env
4) Failed to open Remote env
5) Other operational errors that we don't handle explicitly -> anyhow::Error, not a fan of it but its easy enough

The idea is that we shouldn't just forward opaque errors but separate the context earlier if we can

Reply:
I think we can pass more context through if we need it at some point